### ISSUE TITLE
Implement `PositionManager::fill_route`

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/penumbra#2337]
This can happen in parallel with the other routing-related tasks, since it has a known interface and happens separately from the spill phase.

**Comment by conorsch** [comment]
Landed in #2369! @erwanor please open a follow-up ticket to track deeper testing, and augmenting the scenarios it supports.
